zenodo/modules/stats/exporters.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2018 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo stats exporters."""
+
+import json
+import requests
+
+from dateutil.parser import parse as dateutil_parse
+from flask import current_app
+from elasticsearch_dsl import Search
+from invenio_cache import current_cache
+from invenio_search import current_search_client
+
+from zenodo.modules.stats.utils import chunkify, fetch_record
+
+
+class PiwikExporter:
+    """Events exporter."""
+
+    def run(self, start_date=None, end_date=None, update_bookmark=True):
+        """Run export job."""
+
+        start_date = dateutil_parse(start_date) if start_date else None
+        end_date = dateutil_parse(end_date) if end_date else None
+
+        if start_date is None:
+            bookmark = current_cache.get('piwik_export:bookmark')
+            start_date = dateutil_parse(bookmark) if bookmark else None
+
+        time_range = {}
+        if start_date is not None:
+            time_range['gte'] = start_date.replace(microsecond=0).isoformat()
+        if end_date is not None:
+            time_range['lte'] = end_date.replace(microsecond=0).isoformat()
+
+        events = Search(
+            using=current_search_client,
+            index='events-stats-*'
+        ).filter(
+            'range', timestamp=time_range
+        ).sort(
+            {'timestamp': {'order': 'asc'}}
+        ).params(preserve_order=True).scan()
+
+        url = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'].get('url', None)
+        token_auth = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'] \
+            .get('token_auth', None)
+
+        chunk_size = 85
```

Comment:
Maybe add a comment here explaining why it's 85 (i.e. `[max piwik payload size = 64k] / [max querystring size = 750]`. Even better you can move it in the `ZENODO_STATS_PIWIK_EXPORTER` config (including the comment), so that we can change it in case we see any troubles.